--- origsrc/mailutils-3.2/libmailutils/auth/system.c	2017-01-01 08:47:08.000000000 -0500
+++ src/mailutils-3.2/libmailutils/auth/system.c	2017-04-22 09:51:12.303260200 -0400
@@ -119,9 +119,9 @@ mu_authenticate_system (struct mu_auth_d
 			void *func_data MU_ARG_UNUSED,
 			void *call_data)
 {
+#ifdef HAVE_SHADOW_H
   const struct mu_auth_data *auth_data = key;
 
-#ifdef HAVE_SHADOW_H
   char *pass = call_data;
   
   if (auth_data)
--- origsrc/mailutils-3.2/libmailutils/base/copyfile.c	2017-03-11 09:06:21.000000000 -0500
+++ src/mailutils-3.2/libmailutils/base/copyfile.c	2017-04-22 09:48:17.526749800 -0400
@@ -163,7 +163,7 @@ copy_regular_file (const char *srcpath,
       rc = mu_stream_ioctl (dst, MU_IOCTL_TRANSPORT, MU_IOCTL_OP_GET, trans);
       if (rc == 0)
 	{	    
-	  if (fchmod ((int) trans[0], mode))
+	  if (fchmod ((intptr_t) trans[0], mode))
 	    {
 	      mu_debug (MU_DEBCAT_STREAM, MU_DEBUG_ERROR,
 			(_("%s: cannot chmod: %s"),
@@ -193,7 +193,7 @@ copy_regular_file (const char *srcpath,
 
 	      if (gid != -1)
 		{
-		  if (fchown ((int) trans[0], uid, gid))
+		  if (fchown ((intptr_t) trans[0], uid, gid))
 		    {
 		      mu_debug (MU_DEBCAT_STREAM, MU_DEBUG_ERROR,
 				(_("%s: cannot chown to %lu.%lu: %s"),
--- origsrc/mailutils-3.2/libmailutils/tests/globtest.c	2017-01-01 08:47:10.000000000 -0500
+++ src/mailutils-3.2/libmailutils/tests/globtest.c	2017-04-22 10:12:46.074309900 -0400
@@ -103,7 +103,7 @@ main (int argc, char **argv)
 
 	      for (j = 0; j < nmatch; j++)
  		printf ("%02zu: %.*s\n", j,
-			matches[j].rm_eo - matches[j].rm_so,
+			(int) (matches[j].rm_eo - matches[j].rm_so),
 			a + matches[j].rm_so);
 	    }
 	}
