--- origsrc/mailutils-3.2/lib/Makefile.am	2017-03-22 16:36:18.437780100 -0400
+++ src/mailutils-3.2/lib/Makefile.am	2017-03-22 17:35:49.250616100 -0400
@@ -36,6 +36,34 @@ libmuaux_la_SOURCES = \
  mu_umaxtostr.c\
  mu_umaxtostr.h
 libmuaux_la_LIBADD=gnu/libgnu.la
+
+## The following list is taken from ../libmailutils/Makefile.am
+libmuaux_la_LIBADD += \
+ @MU_COMMON_LIBRARIES@\
+ ../libmailutils/auth/libauth.la\
+ ../libmailutils/base/libbase.la\
+ ../libmailutils/address/libaddress.la\
+ ../libmailutils/sockaddr/libsockaddr.la\
+ ../libmailutils/cidr/libcidr.la\
+ ../libmailutils/cfg/libcfg.la\
+ ../libmailutils/cli/libcli.la\
+ ../libmailutils/datetime/libdatetime.la\
+ ../libmailutils/diag/libdiag.la\
+ ../libmailutils/filter/libfilter.la\
+ ../libmailutils/imapio/libimapio.la\
+ ../libmailutils/list/liblist.la\
+ ../libmailutils/mailbox/libmailbox.la\
+ ../libmailutils/mailer/libmailer.la\
+ ../libmailutils/mime/libmime.la\
+ ../libmailutils/msgset/libmsgset.la\
+ ../libmailutils/opt/libopt.la\
+ ../libmailutils/property/libproperty.la\
+ ../libmailutils/server/libserver.la\
+ ../libmailutils/string/libstring.la\
+ ../libmailutils/stream/libstream.la\
+ ../libmailutils/stdstream/libstdstream.la\
+ ../libmailutils/url/liburl.la
+
 libmuaux_la_LDFLAGS = -no-undefined -version-info @VI_CURRENT@:@VI_REVISION@:@VI_AGE@
 
 libmutcpwrap_a_SOURCES = tcpwrap.c
