--- origsrc/mailutils-3.2/maidag/Makefile.am	2017-01-01 08:47:12.000000000 -0500
+++ src/mailutils-3.2/maidag/Makefile.am	2017-03-19 13:20:08.096830600 -0400
@@ -53,7 +53,8 @@ maidag_LDADD = \
  @GUILE_LIBS@\
  @PYTHON_LIBS@\
  @DBMLIBS@\
- @MU_TCPWRAP_LIBRARIES@
+ @MU_TCPWRAP_LIBRARIES@\
+ ../libmailutils/sockaddr/libsockaddr.la
 
 AM_CPPFLAGS = -I${top_srcdir} @MU_APP_COMMON_INCLUDES@ @GUILE_INCLUDES@ \
  @PYTHON_INCLUDES@
